frontend/src/components/CuisineFilter.tsx

```diff
@@ -0,0 +1,94 @@
+import { cuisineList } from '@/config/restaurant-options-config';
+import { Label } from './ui/label';
+import { Check, ChevronDown, ChevronUp } from 'lucide-react';
+import { ChangeEvent } from 'react';
+import { Button } from './ui/button';
+
+type Props = {
+  onChange(cuisines: string[]): void;
+  selectedCuisines: string[];
+  isExpanded: boolean;
+  onExpandClick(): void;
+};
+
+const CuisineFilter = ({
+  onChange,
+  selectedCuisines,
+  isExpanded,
+  onExpandClick,
+}: Props) => {
+  const handleCuisinesReset = () => {
+    onChange([]);
+  };
+
+  const handleCuisinesChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const clickedCuisine = event.target.value;
+    const isChecked = event.target.checked;
+    const newCuisinesLint = isChecked
+      ? [...selectedCuisines, clickedCuisine]
+      : selectedCuisines.filter(
+          (cuisine: string) => cuisine !== clickedCuisine
+        );
+
+    onChange(newCuisinesLint);
+  };
+
+  return (
+    <>
+      <div className="flex justify-between items-center px-2">
+        <div className="text-md font-semibold mb-2">Filter by Cuisine</div>
+        <div
+          onClick={handleCuisinesReset}
+          className="text-md font-semibold mb-2 underline cursor-pointer text-blue-500"
+        >
+          Reset Filters
+        </div>
+      </div>
+
+      <div className="space-y-2 flex flex-col">
+        {cuisineList
+          .slice(0, isExpanded ? cuisineList.length : 7)
+          .map((cuisine) => {
+            const isSelected = selectedCuisines.includes(cuisine);
+            return (
+              <div className="flex">
+                <input
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/41#issuecomment-2410866501>#e5d62a</a><span> - 10/14/2024, 11:04 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The custom checkbox implementation may not be fully accessible. The input is hidden, which could cause issues for screen readers and keyboard navigation.
</div>
<div id="fix">
<b>Fix:</b> Implement an accessible custom checkbox by using proper ARIA attributes and ensuring keyboard navigability. Use a visually hidden class instead of `display: none` to hide the native checkbox while keeping it accessible to screen readers.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -55,7 +55,7 @@
                  <input
                    id={`cuisine_${cuisine}`}
                    type="checkbox"
 -                  className="hidden"
 +                  className="sr-only"
                    value={cuisine}
                    checked={isSelected}
                    onChange={handleCuisinesChange}
 @@ -63,7 +63,8 @@
                  <Label
                    htmlFor={`cuisine_${cuisine}`}
                    className={`flex flex-1 item-center cursor-pointer text-sm rounded-full px-4 py-2 font-semibold ${
                      isSelected
                        ? 'border border-green-600 text-green-600'
                        : 'border border-slate-300'
 -                  }`}
 +                  }`}
 +                  aria-checked={isSelected}
                  >
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged